.github/workflows/main.yml

```diff
@@ -32,40 +48,29 @@ jobs:
           cmake -DENABLE_TESTS:BOOL=ON -DCMAKE_BUILD_TYPE=${OROCOS_KDL_BUILD_TYPE} ./..
           make
           sudo make install
-      - name: Build PyKDL Python 2
+      - name: Build PyKDL
         run: |
           cd python_orocos_kdl
-          mkdir build2
-          cd build2
-          export ROS_PYTHON_VERSION=2
-          cmake -DCMAKE_BUILD_TYPE=${OROCOS_KDL_BUILD_TYPE} ./..
-          make
-          sudo make install
-      - name: Build PyKDL Python 3
-        run: |
-          cd python_orocos_kdl
-          mkdir build3
-          cd build3
-          export ROS_PYTHON_VERSION=3
+          mkdir build
+          cd build
           cmake -DCMAKE_BUILD_TYPE=${OROCOS_KDL_BUILD_TYPE} ./..
           make
           sudo make install
       - name: Update LD_LIBRARY_PATH
         run: |
-          export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib
+          export LD_LIBRARY_PATH=/usr/local/lib${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
           sudo ldconfig
+          echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH" >> $GITHUB_ENV
       - name: Test orocos_kdl
         run: |
           cd orocos_kdl/build
           make check
-      - name: Test PyKDL Python 2
-        run: |
-          cd python_orocos_kdl
-          python2 tests/PyKDLtest.py
-      - name: Test PyKDL Python 3
+      - name: Test PyKDL
         run: |
           cd python_orocos_kdl
-          python3 tests/PyKDLtest.py
+          python_version_short=$(python -c "import sys; print('{}.{}'.format(sys.version_info[0], sys.version_info[1]))")
+          export PYTHONPATH=/usr/local/lib/python${python_version_short}/dist-packages${PYTHONPATH:+:${PYTHONPATH}}
+          python tests/PyKDLtest.py
```

Comment:
Same here: `/usr/local/lib/pythonX/dist-packages` is already searched by default without the need to patch `PYTHONPATH`?

```sh
$ python -c "import sys; print('\n'.join(sys.path));"
/usr/lib/python2.7/lib-tk
/usr/lib/python2.7/lib-old
/usr/lib/python2.7/lib-dynload
/usr/local/lib/python2.7/dist-packages
/usr/lib/python2.7/dist-packages
/usr/lib/python2.7/dist-packages/gtk-2.0

$ python3 -c "import sys; print('\n'.join(sys.path));"
/usr/lib/python38.zip
/usr/lib/python3.8
/usr/lib/python3.8/lib-dynload
/usr/local/lib/python3.8/dist-packages
/usr/lib/python3/dist-packages
```

That may not be the case for the Python interpreter installed by [action/setup-python@v2](https://github.com/actions/setup-python).

Reply:
Indeed the path which we install to is not in `sys.path` in the interpreter of the setup-python. Therefore this is needed.